core/src/main/java/org/apache/iceberg/OffsetsAwareSplitScanTaskIterator.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+
+/**
+ * An iterator that splits tasks using split offsets such as row group offsets in Parquet.
+ *
+ * @param <T> the Java type of tasks produced by this iterator
+ */
+abstract class OffsetsAwareSplitScanTaskIterator<T extends ScanTask> implements Iterator<T> {
+  private final T parentTask;
+  private final List<Long> offsets;
+  private final List<Long> splitSizes;
+  private int sizeIdx = 0;
```

Comment:
nit: is `splitIndex` more accurate?

Reply:
I copied this code as it was. From what I see, it looks like an index to the size of this iterator that is used to access both offsets and split sizes. Let me do some renames to make it a bit cleaner.

Reply:
I changed the code in `next` a little bit so hopefully the purpose of `sizeIndex` is a bit clearer now.

Reply:
>  it is an index of the size of this iterator that is used to access both offsets and split sizes lists

That is exactly where I found it a little confusing to be named as `sizeIndex`. This index is used to construct a split from the indexed offsets and sizes lists.

```
    long offset = offsets.get(sizeIndex);
    long splitSize = splitSizes.get(sizeIndex);
    sizeIndex += 1; // create 1 split per offset
    return createSplitTaskFunc.apply(parentTask, offset, splitSize);
```

But this is a very minor thing.

Reply:
Yeah, I see what you mean now. I don't know what was the original interpretation but I can switch to `splitIndex`.

Reply:
Switched.